android-mvvm-kotlin/src/main/java/com/manaschaudhari/android_mvvm/MvvmRxExtensions.kt

```diff
@@ -0,0 +1,60 @@
+package com.manaschaudhari.android_mvvm
+
+import android.databinding.*
+import android.support.annotation.CheckResult
+import io.reactivex.Observable
+import io.reactivex.annotations.CheckReturnValue
+import io.reactivex.disposables.Disposable
+import io.reactivex.subjects.PublishSubject
+
+/**
+ * A set of convenience extension functions for Kotlin.
+ *
+ * Example usage:
+ * someObservableField.asObservable().....
+ * Created by Gabor Szanto on 2017. 03. 24..
+ */
+
+/**
+ * From ObservableField to Observable
+ */
+fun <T> ObservableField<T>.asObservable(): Observable<T> = FieldUtils.toObservable(this)
+
+fun <T> ObservableInt.asObservable(): Observable<Int> = FieldUtils.toObservable(this)
+fun <T> ObservableFloat.asObservable(): Observable<Float> = FieldUtils.toObservable(this)
+fun <T> ObservableDouble.asObservable(): Observable<Double> = FieldUtils.toObservable(this)
+fun <T> ObservableLong.asObservable(): Observable<Long> = FieldUtils.toObservable(this)
+fun <T> ObservableList<T>.asObservable(): Observable<List<T>> = FieldUtils.toObservable(this)
+
+/**
+ * From Observable to ObservableField
+ */
+fun <T> Observable<T>.asField(): ObservableField<T> = FieldUtils.toField(this)
+
+@CheckResult
+@CheckReturnValue
+fun <T> Observable<T>.bindTo(field: ObservableField<T>): Disposable = FieldUtils.bindTo(this, field)
+
+/**
+ * Helper method for creating an Observable<Boolean> that emits true when the specified Observables emit something.
+ * This is useful for creating a loading Observable, that emits true if a loader should be visible, and false when
+ * it should be invisible. Finish the call with [endLoading].
+ *
+ * Example:
+ * val loading = startLoading(button1Tap, button2Tap).endLoading(response1, response2)
```

Comment:
Interesting approach. It is very likely that `response1` is created by applying `flatMap` on `button1Tap`. Can we create a complete example? Perhaps we can come up with a custom operator OR `transform`, something like:
```
(result, loading) = onButtonTap.load {
  return api.loadSomething()
}
```